packages/wdio-reporter/src/index.js

```diff
@@ -183,11 +184,68 @@ export default class WDIOReporter extends EventEmitter {
         return true
     }
 
+    /**
+     * Returns everything worth reporting from a suite
+     * @param  {Object}    suite  test suite containing tests and hooks
+     * @return {Object[]}         list of events to report
+     */
+    getEventsToReport (suite) {
+        return [
+            /**
+             * report all tests
+             */
+            ...suite.tests,
+            /**
+             * and only hooks that failed
+             */
+            ...suite.hooks
+                .filter((hook) => Boolean(hook.error))
+        ]
+    }
+
+    /**
+     * Get display for failed tests, e.g. stack trace
+     * @return {Array} Stack trace output
+     */
+    getFailureDisplay () {
```

Comment:
Does this belong in wdio-reporter? It seems like each reporter should be responsible for determining the output format. This seems like it belongs in wdio-dot-reporter rather than the base reporter (which should be output format agnostic)